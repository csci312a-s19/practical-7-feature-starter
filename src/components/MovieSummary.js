import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import StarRating from './StarRating';

const Summary = styled.div`
  margin: 20px;
`;

const Title = styled.p`
  font-weight: bold;
  margin: 0px;
`;
Title.displayName = 'MovieTitle';

const Year = styled.p`
  margin: 0px;
  color: #999999;
`;

const TMDBScore = styled.p`
  margin: 0px;
  font-size: smaller;
`;

function MovieSummary(props) {
  return (
    <Summary>
      <Title onClick={() => props.onClick(props.id)}>{props.title}</Title>
      <StarRating
        rating={props.rating}
        setRating={rating => {
          props.setRatingFor(props.id, rating);
        }}
      />
      <Year>({props.release_date.slice(0, 4)})</Year>
      <TMDBScore>TMDB score: {props.vote_average}</TMDBScore>
    </Summary>
  );
}

MovieSummary.propTypes = {
  title: PropTypes.string.isRequired,
  release_date: PropTypes.string.isRequired,
  rating: StarRating.propTypes.rating, // eslint-disable-line react/no-typos
  vote_average: PropTypes.number.isRequired,
  id: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
  setRatingFor: PropTypes.func.isRequired,
};

MovieSummary.defaultProps = {
  rating: StarRating.defaultProps.rating,
};

export default MovieSummary;
