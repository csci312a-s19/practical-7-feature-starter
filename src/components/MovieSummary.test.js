import React from 'react';
import { shallow } from 'enzyme';

import MovieSummary from './MovieSummary';

const movie = {
  id: 135397,
  overview: 'Twenty-two years after the events of Jurassic Park...',
  release_date: '2015-06-12',
  poster_path: '/jjBgi2r5cRt36xF6iNUEhzscEcb.jpg',
  title: 'Jurassic World',
  vote_average: 6.9,
  genre_ids:[ 28, 12],
};

describe('MovieSummary', () => {
  test('MovieSummary snapshot', () => {
    const comp = shallow(
      <MovieSummary {...movie} onClick={jest.fn} setRatingFor={jest.fn} />
    );
    expect(comp).toMatchSnapshot();
  });
});
