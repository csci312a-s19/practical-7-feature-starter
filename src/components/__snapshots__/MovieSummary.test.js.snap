// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSummary MovieSummary snapshot 1`] = `
<styled.div>
  <MovieTitle
    onClick={[Function]}
  >
    Jurassic World
  </MovieTitle>
  <StarRating
    rating={0}
    setRating={[Function]}
  />
  <styled.p>
    (
    2015
    )
  </styled.p>
  <styled.p>
    TMDB score: 
    6.9
  </styled.p>
</styled.div>
`;
